tests/communication/test_semeio_script.py

```diff
@@ -0,0 +1,204 @@
+import json
+import logging
+import os
+import pytest
+import sys
+
+from semeio.communication import SemeioScript
+
+if sys.version_info >= (3, 3):
+    from unittest.mock import Mock
+else:
+    from mock import Mock
+
+
+def test_semeio_script_publish(tmpdir):
+    tmpdir.chdir()
+    enspath = "enspath"
+    ert = Mock(
+        resConfig=Mock(
+            return_value=Mock(model_config=Mock(getEnspath=Mock(return_value=enspath)))
+        )
+    )
+
+    namespace = "arbitrary_data"
+    data = [1, 2, 3, 4, "This is a very important string"]
+
+    class MySuperScript(SemeioScript):
+        def run(self, *args):
+            self.reporter.publish(namespace, data)
+
+    my_super_script = MySuperScript(ert)
+    my_super_script.run()
+
+    expected_outputfile = os.path.join(
+        enspath, "reports", MySuperScript.__name__, namespace + ".json"
+    )
+
+    with open(expected_outputfile) as f:
+        published_data = json.load(f)
+
+    assert data == published_data
+
+
+def test_semeio_script_logging(tmpdir):
+    tmpdir.chdir()
+    enspath = "enspath"
+    ert = Mock(
+        resConfig=Mock(
+            return_value=Mock(model_config=Mock(getEnspath=Mock(return_value=enspath)))
+        )
+    )
+
+    msg = "My logging msg"
+
+    class MySuperScript(SemeioScript):
+        def run(self, *args):
```

Comment:
shouldnt this also execute the run method of SemeioScript? `super(MySuperScript, self).run() or something?

Reply:
Well, should and could and all of that 🤷 I just wanted to put myself in-between an implementation of an `ErtScript` and `ERT` itself in such a manner that any workflow could start inheriting `SemeioScript` instead of `ErtScript` and would continue to function. Ert will call the `run`-method, while `ErtScript` does in fact not implement `run`. So, there is an expectancy that the child class will implement it. And no child class currently expects there to be a `super.run`-method. Hence, I injected myself in-between by wrapping the `run`-method of the child in the `__init__`.